pkg/prometheus/promcfg.go

```diff
@@ -370,30 +371,30 @@ func (cg *ConfigGenerator) addAuthorizationToYaml(
 	return cg.WithMinimumVersion("2.26.0").WithKeyVals("component", strings.Split(assetStoreKey, "/")[0]).AppendMapItem(cfg, "authorization", authCfg)
 }
 
-func buildExternalLabels(p *v1.Prometheus) yaml.MapSlice {
+func buildExternalLabels(cpf v1.CommonPrometheusFields, name, namespace string) yaml.MapSlice {
```

Comment:
Do we have a reason not to make this function a method of the struct `ConfigGenerator`? This would allow us to do something like the following no? Or am I missing something 🤔 
```suggestion
func (cg *ConfigGenerator) buildExternalLabels() yaml.MapSlice {
```

Reply:
Very good point! I don't think there is any special reason 🤔